### ISSUE TITLE
Can not set on-chain fees in regtest mode

### ISSUE THREAD

**Issue by s-tikhomirov** [ElementsProject/lightning#3406]
### Issue and Steps to Reproduce

I'm trying to open a channel from a c-lightning node to an Eclair node on my local machine on regtest and get this error:
```
{
   "code": -1,
   "message": "They sent error channel 21c19986a601e733e838dc4ac0930aa5cb54b8509ccfe7c7daa2e0f32f197e9f: local/remote feerates are too different: remoteFeeratePerKw=253 localFeeratePerKw=45000"
}
```
Apparently, the Eclair node doesn't like the huge difference in the fee rates (Eclair's rate is 45000, c-lightning's rate is 253). I tried to make Eclair ignore this but didn't succeed. Now I want to set c-lightning's fees to a higher level to be on par with Eclair's. I set `commit-fee=1000` expecting the feerate to raise 10x, but I get the same error with the same value (253).

The fee estimation seems to be failing, but on regtest this is understandable, I suppose?..
```
2020-01-09T15:38:06.266Z UNUSUAL lightningd: Unable to estimate CONSERVATIVE/2 fee
2020-01-09T15:38:06.277Z UNUSUAL lightningd: Unable to estimate ECONOMICAL/4 fee
2020-01-09T15:38:06.287Z UNUSUAL lightningd: Unable to estimate ECONOMICAL/100 fee
2020-01-09T15:38:36.297Z UNUSUAL lightningd: Unable to estimate CONSERVATIVE/2 fee
```
How do I set higher on-chain fees on regtest?

### `getinfo` output
```
{
   "id": "03068d9ac7f6973cbdb7212f2598afee2db14becd3d31e937659b0c09cee9f78f2",
   "alias": "ANGRYMAESTRO",
   "color": "03068d",
   "num_peers": 0,
   "num_pending_channels": 0,
   "num_active_channels": 0,
   "num_inactive_channels": 0,
   "address": [],
   "binding": [
      {
         "type": "ipv4",
         "address": "127.0.0.1",
         "port": 9737
      }
   ],
   "version": "v0.8.0-40-g899f5de",
   "blockheight": 651,
   "network": "regtest",
   "msatoshi_fees_collected": 0,
   "fees_collected_msat": "0msat",
   "lightning-dir": "/home/sergei/.lightning/regtest"
}
```

**Comment by darosior** [comment]
> The fee estimation seems to be failing, but on regtest this is understandable, I suppose?..

Yep

> How do I set higher on-chain fees on regtest?

We gather the feerate from `bitcoind`, so `bitcoin-cli settxfee 4500` should make Eclair happy. That's weird they default to a feerate that huge (especially for regtest), though.

**Comment by s-tikhomirov** [comment]
> bitcoin-cli settxfee 4500 should make Eclair happy
Thanks for the advice, but, surprisingly, it doesn't. `bitcoin-cli settxfee 4500` returned `true, but I get exactly the same error on channel creation. Maybe the problem is that the new fee rate must be included in a channel update and propagated to the Eclair node?

**Comment by ZmnSCPxj** [comment]
> Maybe the problem is that the new fee rate must be included in a channel update and propagated to the Eclair node?

This is not relevant to this case --- this is a disagreement with `update_fee`, not `channel_announcement` or `channel_update`.

What you may need to do would be to wait for C-Lightning to poll the feerate from its `bitcoind` *before* you fund the channel from C-Lightning to Ecliar,  You could monitor this with the undocumented `feerate` command.

**Comment by s-tikhomirov** [comment]
> monitor this with the undocumented feerate command

How do I do that?

```
lightning-cli feerate
{
   "code": -32601,
   "message": "Unknown command 'feerate'"
}
```

**Comment by ZmnSCPxj** [comment]
Sorry that is `feerates` with an `s`.  Crummy undocumented commands....

**Comment by s-tikhomirov** [comment]
Ok, how it requires some `style` option:
```
lightning-cli feerates
{
   "code": -32602,
   "message": "missing required parameter: style"
}
```

I've tries to guess based on [source](https://github.com/ElementsProject/lightning/blob/master/lightningd/chaintopology.c) but wasnt successful:
```
$ lightning-cli feerates --style=FEERATE_PER_KSIPA
lightning-cli: --style=FEERATE_PER_KSIPA: unrecognized option
$ lightning-cli feerates --style=perkw
lightning-cli: --style=perkw: unrecognized option
$ lightning-cli feerates --style=satoshi-per-kw
lightning-cli: --style=satoshi-per-kw: unrecognized option
```

What should I use in the `style` option? (And what is KSIPA, by the way?)

**Comment by darosior** [comment]
`perkw`

**Comment by darosior** [comment]
If you want to use named arguments, you need to specify it explicitly to the cli:
```
lightning-cli -named feerates style=perkw
```

**Comment by darosior** [comment]
But we poll the feerate at start, so doesn't restarting `lightningd` afterwards fixes it ?

**Comment by s-tikhomirov** [comment]
```
$ lightning-cli -named feerates style=perkw
lightning-cli: -named: unrecognized option
```

Restarting doesn't seem to change anything either.

**Comment by darosior** [comment]
```
lightning-cli -k feerates style=perkw
```
*Sorry `-named` is bitcoin-cli's one..

**Comment by s-tikhomirov** [comment]
`lightning-cli -k feerates style=perkw` works!

Indeed, the default tx fee on regtest is 0, which can be checked with `bitcoin-cli getwalletinfo` (`"paytxfee": 0.00000000`). I changed it with `bitcoin-cli settxfee 0.00001` (I also tried other larger values later), did a bunch of transactions and mined some blocks:

```
bitcoin-cli sendtoaddress "2MyRcdBCtdeuYj2WcvfiypzL8NW68bx6zpi" 0.1
bitcoin-cli sendtoaddress "2MyRcdBCtdeuYj2WcvfiypzL8NW68bx6zpi" 0.1
bitcoin-cli sendtoaddress "2MyRcdBCtdeuYj2WcvfiypzL8NW68bx6zpi" 0.1
bitcoin-cli generatetoaddress 1 `bitcoin-cli getnewaddress`
```

After this, the fee estimations changed (though I still can not see the pattern):
```
$ bitcoin-cli estimatesmartfee 1
{
  "feerate": 0.00020000,
  "blocks": 2
}
```

The c-lightning's fee estimations changed as well (earlier `urgent`, `normal`, and `slow` were all equal to 253):
```
$ lightning-cli -k feerates style=perkw
{
   "perkw": {
      "urgent": 5000,
      "normal": 1250,
      "slow": 1250,
      "min_acceptable": 253,
      "max_acceptable": 4294967295
   },
   "onchain_fee_estimates": {
      "opening_channel_satoshis": 877,
      "mutual_close_satoshis": 841,
      "unilateral_close_satoshis": 2990
   }
}
```
However, the channel opening returned the same error, but with a different number:
```
$ lightning-cli fundchannel 03b66cddd29beb92f9fbdce91924cde26904ba7f5c44ddf73c4d3e0379541b3987 10000000
{
   "code": -1,
   "message": "They sent error channel 56045c65b6eb04c75dd4e4deae54d57f29ac7df7a0b052bbdf18aeda1393d510: local/remote feerates are too different: remoteFeeratePerKw=1250 localFeeratePerKw=45000"
}
```

And now comes the punchline: when I tried to raise the maximum feerate mismatch for Eclair, I messed up with the config and did not actually change it. Now I got is right (`eclair.on-chain-fees.max-feerate-mismatch=3.0`), and the channel did open. Now I'm in a strange situation when c-lightning counts Eclair as connected, but Eclair displays c-lightning as disconnected, but that's a different story.

TLDR for the future readers of this thread: the feerate on regtest is 0 by default, if you want to change it, raise it in `bitcoin-cli`, send a few transactions, and mine a few blocks: c-lightning will update its fee estimation.
